packages/react-silk-ui/src/components/Masonry.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react'
+import { View, Text, Image, TouchableWithoutFeedback } from 'react-native'
+
+export interface Props {
+  images: string[]
+  col?: number
+  onPress?: (arg0: { url: string; height: number; width: number }) => void
+}
+
+const defaultProps = {
+  col: 3,
+  onPress: () => {},
+}
+
+interface State {
+  width: number
+  columns: {
+    height: number
+    images: { url: string; height: number; width: number }[]
+  }[]
+}
+
+const COLUMN_WIDTH = 300
+
+export class Masonry extends React.Component<Props, State> {
+  static defaultProps = defaultProps
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      width: 0,
+      columns: [],
+    }
+  }
+
+  async componentDidMount() {
+    const columns = []
+    const { col } = this.props
+    for (let i = 0; i < 3; i++) {
+      columns.push({
+        height: 0,
+        images: [],
+      })
+    }
+    await this.setState({ columns })
+    await this.props.images.forEach(img => {
+      Image.getSize(
+        img,
+        (width, height) => {
+          const index = this.minColIndex()
+          const cols = this.state.columns
+          const imgHeight = (height * COLUMN_WIDTH) / width
+          cols[index].height = cols[index].height + imgHeight
+          cols[index].images.push({
+            url: img,
+            height: imgHeight,
+            width: COLUMN_WIDTH,
+          })
+          this.setState({ columns: cols })
+        },
+        err => console.log(err),
+      )
+    })
+  }
+
+  minColIndex() {
+    return this.state.columns.reduce(
+      (minIndex, current, index, arr) =>
+        current.height < arr[minIndex].height ? index : minIndex,
+      0,
+    )
+  }
+
+  onLayout = (event: any) => {
+    const { width } = event.nativeEvent.layout
+    this.setState({ width })
+  }
+
+  columnWidth() {
+    return this.state.width / 3 - 10
+  }
+
+  render() {
+    const onPress = this.props.onPress!
```

Comment:
onPress truyền vào là optional, thì nên check, nếu không có onPress thì nên thay cái container thành View, chứ nếu để nút bấm thật ra cũng kì kì, vì thật ra onPress vẫn được call, chỉ có cái là call for nothing, check cái component Button của a làm để biết thêm nhé